lib/kubeclient/common.rb

```diff
@@ -113,9 +115,7 @@ def handle_exception
       rescue JSON::ParserError
         {}
       end
-      err_message = json_error_msg['message'] || e.message
-      error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
-      raise error_klass.new(e.http_code, err_message, e.response)
+      http_err(json_error_msg, e)
```

Comment:
I find it unclear from the "http_err" name what it will do, I'd expect it to just return some kind of err.
Perhaps "raise_http_err"?
[bikeshedding, your call]